### PR TITLE
PCIE device wake up enhancement

### PR DESCRIPTION
**Pull request by mauricema**
This is a follow-up to the previous WOL commit. The previous
implementation is more LeafHill specific. This patch further
added configuration data to control how wake up signal is mapped
into each PCIE root port. And the ASL code has been adjusted to
utilize that info so that the code can be more generic for all
other platforms.

The current default configuration for PCIE wake signal is aligned
with LeafHill CRB board. For other borads, to enable it properly,
it is required to override the wake signal configuration using DLT.

Test has been done on LeafHill to do WOL with yocto image. It worked
as expected.

Signed-off-by: Maurice Ma <maurice.ma@intel.com>

### CODE REVIEW

File: Platform/ApollolakeBoardPkg/CfgData/CfgData_PcieRp.dsc [hunk 1]
```diff
@@ -27,11 +27,11 @@
...
   #                                      Power       : Reset
-  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:0 :  0x0021318A  : 0x0021618E}
-  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:1 :  0x0021118A  : 0x0020D18A}
-  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:2 :  0x0000018B  : 0x0020F18A}
+  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:0 :  0x6021318A  : 0x0021618E}
+  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:1 :  0x8021118A  : 0x0020D18A}
+  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:2 :  0x4000018B  : 0x0020F18A}
   # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:3 :  0x0000018B  : 0x0022518A}
-  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:4 :  0x0000018B  : 0x0000018B}
+  # !BSF SUBT:{PCIERP_CTRL_PIN_TMPL:4 :  0x2000018B  : 0x0000018B}
```
Comment (umaplehurst) [inline Platform/ApollolakeBoardPkg/CfgData/CfgData_PcieRp.dsc]: I have double-checked these against the Leaf Hill CRB schematic and all looks valid and correct to me.

File: Platform/ApollolakeBoardPkg/AcpiTables/Dsdt/Sc.asl [hunk 2]
```diff
@@ -126,9 +135,12 @@ scope (\_SB.PCI0) {
...
     }
-    Name(_PRF, 0x08)
-    Include("ScPcie.asl")
-    Method(_PRW, 0) { Return(GPRW(_PRF, 4)) }  // Leaf Hill CRB: GPE enable bit for RP2 is 8, can wake up from S4 state
+
+    #define PCIE_WAKE_GPE_BIT  RPW2
+    #include "ScPcie.asl"
+    #undef  PCIE_WAKE_GPE_BIT
+
+    Method(_PRW, 0) { Return(GPRW(RPW2, 4)) }  //GPE enable bit for RP2 is 8, can wake up from S4 state
```
Comment (umaplehurst) [inline Platform/ApollolakeBoardPkg/AcpiTables/Dsdt/Sc.asl]: This comment doesn't have any relevance any more, should be just "can wakeup from S4 state" like the others?

**Comment by umaplehurst** [comment]
Looks great, I really like it! Tested on my side also and it works great.

**Comment by mauricema** [comment]
@umaplehurst:  Thank you very much for reviewing it!  Updated this commit to address the review comments.

File: Platform/ApollolakeBoardPkg/AcpiTables/Dsdt/Sc.asl (hunk 2 above)
Comment (mauricema) [reply Platform/ApollolakeBoardPkg/AcpiTables/Dsdt/Sc.asl]: Sure,  Fixed it in new commit.

File: Platform/ApollolakeBoardPkg/CfgData/CfgData_PcieRp.dsc (hunk 1 above)
Comment (mauricema) [reply Platform/ApollolakeBoardPkg/CfgData/CfgData_PcieRp.dsc]: Thanks for confirmation.

**Pull request merged**